csvcubed/tests/unit/cube/qb/test_cubeqb_errorvalidation.py

```diff
@@ -698,5 +699,43 @@ def test_attribute_numeric_resources_validation():
     assert len(errors) == 0
 
 
+def test_code_list_concept_identifier_reserved():
+    """
+    Test that if the user tries to define a code-list with a concept which would use the reserved `code-list`
+    identifier then they get a suitable validation error.
+    """
+    data = pd.DataFrame(
+        {
+            "New Dimension": ["A", "B", "C", "Code List"],
+            "Value": [2, 2, 2, 2],
+        }
+    )
+
+    qube = Cube(
+        metadata=CatalogMetadata("Some Qube"),
+        data=data,
+        columns=[
+            QbColumn(
+                "New Dimension",
+                NewQbDimension.from_data("New Dimension", data["New Dimension"]),
+            ),
+            QbColumn(
+                "Value",
+                QbSingleMeasureObservationValue(
+                    NewQbMeasure("Some Measure"),
+                    NewQbUnit("Some Unit"),
+                ),
+            ),
+        ],
+    )
+    errors = qube.validate() + validate_qb_component_constraints(qube)
+    assert_num_validation_errors(errors, 1)
+    error = errors[0]
+    assert isinstance(error, ReservedUriValueError)
+    assert error.csv_column_name == "New Dimension"
+    assert error.reserved_identifier == "code-list"
+    assert error.conflicting_values == ["Code List"]
```

Comment:
I think this means that if a user wanted a code list with "code list" in it they could create a label of "code list" provided the value was something like "cd list", right?

Reply:
My initial answer was that as long as you provide a `uri_safe_override` of something which isn't `code-list`, you should be fine. 

But then I noticed we aren't actually checking any of the `uri_safe_override` values when we're enforcing uniqueness. Since the (python) user can provide such overrides, it's actually *those* values, not just the `uri_safe(label)` we've been working with so far that we need to ensure are unique.

Ugh.

Reply:
I suspect we'll need to spin this problem out into a separate issue.

Reply:
https://github.com/GSS-Cogs/csvcubed/issues/309

Reply:
Yuk